core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -3185,11 +3185,15 @@ private static int unsetLocalDictForComplexColumns(List<ColumnSchema> allColumns
       ColumnSchema column = allColumns.get(dimensionOrdinal);
       if (column.getNumberOfChild() > 0) {
         dimensionOrdinal++;
-        unsetLocalDictForComplexColumns(allColumns, dimensionOrdinal, column.getNumberOfChild());
+        // Value of dimensionOrdinal returned from recursive function was not considered
```

Comment:
do not give the reason in comment, why this issue was coming, add proper comment